File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0;
        int right = height.length - 1;
        int curWater = 0;
        int mostWater = 0;

        while (left < right) {
            curWater = Math.min(height[left], height[right]) * (right - left);
            mostWater = Math.max(mostWater, curWater);

            if (height[left] < height[right]) left++;
            else right--;
        }

        return mostWater;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.kt
class Solution {
    fun maxArea(height: IntArray): Int {
        var left = 0
        var right = height.size - 1
        var curWater = 0
        var mostWater = 0

        while (left < right) {
            curWater = minOf(height[left], height[right]) * (right - left)
            mostWater = maxOf(mostWater, curWater)

            if (height[left] < height[right]) left++
            else right--
        }

        return mostWater
    }
}